`tvatiteropenarray` test fails with c++

```nim
iterator iterAndZero(a: var openArray[int]): int =
  for i in 0..len(a)-1:
    yield a[i]
    a[i] = 0

var x = [[1, 2, 3], [4, 5, 6]]
for y in iterAndZero(x[0]): write(stdout, $y)
#OUT 123

write stdout, "\n"
```

```
nimcache/@mtitervaropenarray.nim.cpp:90:59: error: member reference base type 'tyArray__Bd4h7Ocx9bGTvrKzPIWNlHw' (aka 'long[3]') is not a structure or union
                        colontmp_ = (NI)(x__titervaropenarray_12[(((NI) 0))- 0].Field1 - ((NI) 1));
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
nimcache/@mtitervaropenarray.nim.cpp:96:70: error: member reference base type 'tyArray__Bd4h7Ocx9bGTvrKzPIWNlHw' (aka 'long[3]') is not a structure or union
                                        y__titervaropenarray_13 = x__titervaropenarray_12[(((NI) 0))- 0].Field0[i];
                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
nimcache/@mtitervaropenarray.nim.cpp:99:44: error: member reference base type 'tyArray__Bd4h7Ocx9bGTvrKzPIWNlHw' (aka 'long[3]') is not a structure or union
                                        x__titervaropenarray_12[(((NI) 0))- 0].Field0[i] = ((NI) 0);
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
```

```sh
nim cpp tvariteropenarray
```
nim 1.6.6

--- a/compiler/transf.nim
+++ b/compiler/transf.nim
@@ -472,7 +472,8 @@ proc transformYield(c: PTransf, n: PNode): PNode =
 
 proc transformAddrDeref(c: PTransf, n: PNode, kinds: TNodeKinds): PNode =
   result = transformSons(c, n)
-  if c.graph.config.backend == backendCpp or sfCompileToCpp in c.module.flags: return
+  # inlining of 'var openarray' iterators; bug #19977
+  if n.typ.kind != tyOpenArray and (c.graph.config.backend == backendCpp or sfCompileToCpp in c.module.flags): return
   var n = result
   case n[0].kind
   of nkObjUpConv, nkObjDownConv, nkChckRange, nkChckRangeF, nkChckRange64:

--- a/tests/iter/titervaropenarray.nim
+++ b/tests/iter/titervaropenarray.nim
@@ -1,6 +1,6 @@
 discard """
   output: "123"
-  targets: "c"
+  targets: "c cpp"
 """
 # Try to break the transformation pass:
 iterator iterAndZero(a: var openArray[int]): int =

